### PR TITLE
[minigraph] Fix parser on PNG DeviceInterfaceLink Bandwidth

### PR DESCRIPTION
**Pull request by qiluo-msft**
If PNG/DeviceInterfaceLink section provides a Bandwidth, we should consider it with high priority.

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -369,6 +376,7 @@ def parse_xml(filename, platform=None, port_config_file=None):
     devices = None
     hostname = None
     port_speeds = {}
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: -> port_speeds_default

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -369,6 +376,7 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
+    port_speeds1 = {}
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: -> port_speed_png

File: src/sonic-config-engine/tests/simple-sample-graph.xml [hunk 3]
```diff
@@ -181,7 +181,19 @@
...
+    <DeviceInterfaceLinks>
+      <DeviceLinkBase i:type="DeviceInterfaceLink">
+        <ElementType>DeviceInterfaceLink</ElementType>
+        <AutoNegotiation>true</AutoNegotiation>
+        <Bandwidth>1000</Bandwidth>
+        <EndDevice>ARISTA01T1</EndDevice>
+        <EndPort>et1</EndPort>
+        <FlowControl>true</FlowControl>
+        <StartDevice>switch-t0</StartDevice>
+        <StartPort>Ethernet8</StartPort>
```
Comment (lguohan) [inline src/sonic-config-engine/tests/simple-sample-graph.xml]: this should be fortyGigE0/8

File: src/sonic-config-engine/minigraph.py [hunk 4]
```diff
@@ -438,14 +446,20 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
-        ports.setdefault(port_name, {})['speed'] = port_speeds[port_name]
-        if port_speeds[port_name] == '100000':
-            ports.setdefault(port_name, {})['fec'] = 'rs'
+        ports.setdefault(port_name, {})['speed'] = port_speeds_default[port_name]
+    for port_name in port_speed_png:
+        ports.setdefault(port_name, {})['speed'] = port_speed_png[port_name]
+    for port_name, port in ports.items():
+        if port.get('speed') == '100000':
+            port['fec'] = 'rs'
+    for port_name in port_descriptions:
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: duplicated.

File: src/sonic-config-engine/minigraph.py [hunk 5]
```diff
@@ -438,14 +446,20 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
             continue
 
-        ports.setdefault(port_name, {})['speed'] = port_speeds[port_name]
-        if port_speeds[port_name] == '100000':
-            ports.setdefault(port_name, {})['fec'] = 'rs'
+        ports.setdefault(port_name, {})['speed'] = port_speeds_default[port_name]
+    for port_name in port_speed_png:
+        ports.setdefault(port_name, {})['speed'] = port_speed_png[port_name]
+    for port_name, port in ports.items():
+        if port.get('speed') == '100000':
```
Comment (taoyl-ms) [inline src/sonic-config-engine/minigraph.py]: Is it guaranteed that 'speed' will be available for all ports?

File: src/sonic-config-engine/minigraph.py [hunk 6]
```diff
@@ -438,14 +446,20 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
+    for port_name in port_speeds_default:
         # ignore port not in port_config.ini
         if not ports.has_key(port_name):
             continue
 
-        ports.setdefault(port_name, {})['speed'] = port_speeds[port_name]
-        if port_speeds[port_name] == '100000':
-            ports.setdefault(port_name, {})['fec'] = 'rs'
+        ports.setdefault(port_name, {})['speed'] = port_speeds_default[port_name]
+    for port_name in port_speed_png:
```
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply src/sonic-config-engine/minigraph.py]: offline discussed, check the comment for the conclusion. Don't need to treat it as an error here.

---
In reply to: [181198316](https://github.com/Azure/sonic-buildimage/pull/1592#discussion_r181198316) [](ancestors = 181198316)

File: src/sonic-config-engine/minigraph.py (hunk 5 above)
Comment (qiluo-msft) [reply src/sonic-config-engine/minigraph.py]: Reading code, seems not.

---
In reply to: [181201283](https://github.com/Azure/sonic-buildimage/pull/1592#discussion_r181201283) [](ancestors = 181201283)

**Pull request merged**